### DISCUSSION TITLE
Do not update control vars in 'for' loop body

### DISCUSSION THREAD

**Discussion by Steven-Jurado** [nolimits4web/swiper#5680] [Q&A]
![image](https://user-images.githubusercontent.com/60909590/165562179-5037344c-2802-4fee-8cd1-9bd1acec2ec9.png)

Hello, I am sending you this image, it is from a Kiuwan vulnerability report. The security area does not allow me to deploy to production due to this vulnerability.

How can I fix this "vulnerability"? 

![image](https://user-images.githubusercontent.com/60909590/165564082-088cfc6e-ffb4-47b6-a7c9-a1d341d44697.png)

**Comment by MUsamaTahir** [comment]
I came across the same issue but found below use cases in the code:

- Updating the control variable (i) inside the loop
`for ( let i=0 ; i<a.length ; i+=1) { 
 if (special_case) {
 i+=2;
 }
}`

- Used nested loop but the control variable was the same for both loops
`for ( let i=0 ; i<a.length ; i+=1) {
       for ( ; i!== some_condition ; i+=2) {}
 }`

- Used two control variables in a loop and was updating one of them inside the loop
`for ( let i=0 ; i<a.length || j !== 10 ; i+=1) { 
 j = some_data;
}`


There might also be other use cases, so you need to check those in your code.
Else you can post it for further assistance
